### PR TITLE
fix: support "extend type Query"

### PR DESCRIPTION
**Pull request by TYzzt**
https://github.com/Netflix/dgs-intellij-plugin/issues/43
image

### CODE REVIEW

**Comment by TYzzt** [comment]
Hi @berngp , could you please help to review this PR
Thanks!

File: src/main/java/com/netflix/dgs/plugin/services/internal/GraphQLSchemaRegistry.java [hunk 1]
```diff
@@ -37,11 +38,12 @@ public GraphQLSchemaRegistry(Project project) {
...
     Optional<PsiElement> psiForSchemaType(@NotNull PsiElement psiElement, @Nullable String parentType, @Nullable String field) {
-
         TypeDefinitionRegistry registry = getRegistry(psiElement);
-        Optional<ObjectTypeDefinition> objectType = getTypeDefinition(registry, parentType);
-        if (objectType.isPresent()) {
-            Optional<FieldDefinition> schemaField = objectType.get().getFieldDefinitions().stream().filter(f -> f.getName().equals(field)).findAny();
+        List<ObjectTypeDefinition> objectTypes = getTypeDefinitions(registry, parentType);
+        if (!objectTypes.isEmpty()) {
+            var fieldDefinitionList = new ArrayList<FieldDefinition>();
+            objectTypes.stream().map(ObjectTypeDefinition::getFieldDefinitions).forEach(fieldDefinitionList::addAll);
```
Comment (berngp) [inline src/main/java/com/netflix/dgs/plugin/services/internal/GraphQLSchemaRegistry.java:45]: Nit: do we need the array list above? Can't we just collect to list from the stream?

**Comment by berngp** [comment]
Very cool, thanks for the PR!

**Review by berngp — APPROVED** [review]

File: src/main/java/com/netflix/dgs/plugin/services/internal/GraphQLSchemaRegistry.java (hunk 1 above)
Comment (TYzzt) [reply src/main/java/com/netflix/dgs/plugin/services/internal/GraphQLSchemaRegistry.java]: you're right,stream is enough

**Review by berngp — APPROVED** [review]

**Pull request merged**